lib/matplotlib/__init__.py

```diff
@@ -330,8 +330,12 @@ def impl(args, regex, min_ver=None, ignore_exit_code=False):
         message = "Failed to find a Ghostscript installation"
         raise ExecutableNotFoundError(message)
     elif name == "inkscape":
-        # use "no-gui" option for inkscape version detection:
-        info = impl(["inkscape", "-z", "-V"], "^Inkscape ([^ ]*)")
+        try:
+            # use "without-gui" option (only works with inkscape version < 1.0):
+            info = impl(["inkscape", "--without-gui", "-V"], "Inkscape ([^ ]*)")
+        except subprocess.CalledProcessError as _cpe:
```

Comment:
I think this should catch ExecutableNotFoundError?

Reply:
But that's an expected exception from this function?

Reply:
I mean if using inkscape>1 then the first call will raise an ExecutableNotFoundError which we want to catch to try the second call form.  (I think?)

Reply:
But why? That would just raise `ExecutableNotFoundError` the second time too. It's not trying a different executable name or path.

Reply:
I checked and this indeed doesn't work as is with inkscape>=1: in that case the first call to impl raises ExecutableNotFoundError (the unsupported --without-gui flag causes subprocess.check_output to throw a CalledProcessError *which gets converted to ENFError by impl*), and it is *that* exception that you want to catch here to try the second form -- letting any exception thrown by the second form propagate out.

Reply:
> I think this should catch ExecutableNotFoundError?

Apologies, you're right. That's what I get for looking at an older version of the code and trying to do a small change just using the Github online editor... I properly brought this branch down into my IDE and the right thing is to catch the `ExecutableNotFoundError`. Technically, this isn't quite the right error name, since the executable is found, but the argument is invalid, but I'm okay with that if you all are.

Tested the most recent commit on my system with inkscape 0.92.5 and 1.0, and both work (well, 1.0 raises the error due to the `"Matplotlib only supports Inkscape<1.0"` line, but that's expected). I also tested on the system that was raising the `DISPLAY` errors and it works with 0.92.3 and 1.0.